app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
```

Comment:
Why not `return $('#hosts-table tr').map(function(){return $(this).data("host-id")})`?